src/nvim/mark.c

```diff
@@ -1552,3 +1552,92 @@ void mark_mb_adjustpos(buf_T *buf, pos_T *lp)
     }
   }
 }
+
+
+// Add information about mark 'mname' to list 'l'
+static int add_mark(list_T *l, char *mname, pos_T *pos, int bufnr, char *fname)
```

Comment:
Only `fname` param can be set to `NULL`.

Reply:
Sorry, I'm not yet familiar with nvim's function attributes. (that's also the reason for the useless NULL checks, etc.)
Is there some documentation about what the attributes mean?
What do you mean with your comment about `fname`?

Reply:
I can't find the docs for it so get by with the macro definitions in https://github.com/neovim/neovim/blob/master/src/nvim/func_attr.h for now. Enforce nonnull function params via `FUNC_ATTR_NONNULL_ARG`.

Reply:
`l` cannot be null because of tv_list_append_dict.
`mname` is the name of the mark so it is always nonnull even though `tv_dict_add_str`  allows NULL.
`pos->lnum` forces `pos` to be nonnull.
`bufnr` is not a pointer. You can skip this in `FUNC_ATTR_NONNULL_ARG`.
`fname` can be NULL because of `get_buf_local_marks`.

Reply:
Check if some of this parameters can be `const`. https://neovim.io/develop/style-guide.xml#Use_of_const